File: dotnet/samples/FluentCards.Samples/ActionSubmitExecuteSample.cs
using System.Text.Json;

namespace FluentCards.Samples;

/// <summary>
/// Demonstrates creating Submit and Execute actions with custom verbs and data payloads.
/// </summary>
public static class ActionSubmitExecuteSample
{
    /// <summary>
    /// Creates a card with Action.Execute and Action.Submit actions.
    /// </summary>
    /// <returns>An Adaptive Card with Execute and Submit actions.</returns>
    public static AdaptiveCard CreateActionSubmitExecuteCard()
    {
        return new AdaptiveCard
        {
            Version = "1.4",
            Body =
            [
                new TextBlock
                {
                    Text = "welcome to ac 11",
                    Size = TextSize.Large,
                    Weight = TextWeight.Bolder
                },
                new TextBlock
                {
                    Text = "click the buttons below"
                }
            ],
            Actions =
            [
                new ExecuteAction
                {
                    Data = JsonSerializer.SerializeToElement(new { message = "button clicked !!" }),
                    Verb = "testAction",
                    Title = "Test AC Action"
                },
                new SubmitAction
                {
                    Data = JsonSerializer.SerializeToElement(new
                    {
                        msteams = new
                        {
                            type = "task/fetch"
                        }
                    }),
                    Title = "Open Task Module"
                },
                new ExecuteAction
                {
                    Verb = "requestFileUpload",
                    Title = "Request File Upload"
                }
            ]
        };
    }
}

File: node/samples/deno/action_submit_execute_sample.ts
// Deno sample: Action.Submit and Action.Execute with custom verbs and data
// 
// To run: deno run action_submit_execute_sample.ts

import { AdaptiveCardBuilder, TextSize, TextWeight } from 'jsr:@adaptivecards/fluent';
// import { AdaptiveCardBuilder, TextSize, TextWeight } from '../../packages/fluent-cards/src/index.ts';

/** Creates a card with Action.Execute and Action.Submit actions and custom verbs/data. */
export function createActionSubmitExecuteCard() {
  return AdaptiveCardBuilder.create()
    .withVersion('1.4')
    .addTextBlock((tb) =>
      tb
        .withText('welcome to ac 11')
        .withSize(TextSize.Large)
        .withWeight(TextWeight.Bolder),
    )
    .addTextBlock((tb) => tb.withText('click the buttons below'))
    .addAction((a) =>
      a
        .execute('Test AC Action')
        .withData({ message: 'button clicked !!' })
        .withVerb('testAction'),
    )
    .addAction((a) =>
      a
        .submit('Open Task Module')
        .withData({ msteams: { type: 'task/fetch' } }),
    )
    .addAction((a) => a.execute('Request File Upload').withVerb('requestFileUpload'))
    .build();
}

// Run when invoked directly
if (import.meta.main) {
  console.log('=== Action Submit/Execute Sample ===\n');
  console.log('Action Card:', JSON.stringify(createActionSubmitExecuteCard(), null, 2));
}
